pkg/apis/batch/validation/validation.go

```diff
@@ -486,9 +499,115 @@ func validatePodTemplateUpdate(spec, oldSpec batch.JobSpec, fldPath *field.Path,
+			}
+		}
+		if !ptr.Equal(oldJob.Status.CompletionTime, job.Status.CompletionTime) || isConditionTrue(oldJob.Status.Conditions, batch.JobComplete) != isConditionTrue(job.Status.Conditions, batch.JobComplete) {
+			if job.Status.CompletionTime != nil && !isConditionTrue(job.Status.Conditions, batch.JobComplete) {
+				allErrs = append(allErrs, field.Invalid(statusFld.Child("completionTime"), job.Status.CompletionTime, "cannot set completionTime when there is no Complete=True condition"))
+			}
+			if oldJob.Status.CompletionTime == nil && job.Status.CompletionTime == nil && isConditionTrue(job.Status.Conditions, batch.JobComplete) {
+				allErrs = append(allErrs, field.Required(statusFld.Child("completionTime"), "completionTime is required for Complete jobs"))
+			}
+		}
+		if oldJob.Status.StartTime != nil && !ptr.Deref(job.Spec.Suspend, false) {
+			if job.Status.StartTime == nil {
+				allErrs = append(allErrs, field.Required(statusFld.Child("startTime"), "startTime cannot be removed for unsuspended job"))
+			}
+		}
+		if !ptr.Equal(oldJob.Status.StartTime, job.Status.StartTime) || !ptr.Equal(oldJob.Status.CompletionTime, job.Status.CompletionTime) {
+			if job.Status.StartTime != nil && job.Status.CompletionTime != nil && job.Status.CompletionTime.Before(job.Status.StartTime) {
+				allErrs = append(allErrs, field.Invalid(statusFld.Child("completionTime"), job.Status.CompletionTime, "completionTime cannot be set before startTime"))
+			}
+		}
+		// validation when transitioning to terminal phase
+		for _, cType := range []batch.JobConditionType{batch.JobFailed, batch.JobComplete} {
+			if !isConditionTrue(oldJob.Status.Conditions, cType) && isConditionTrue(job.Status.Conditions, cType) {
+				if job.Status.StartTime == nil {
+					allErrs = append(allErrs, field.Required(statusFld.Child("startTime"), fmt.Sprintf("startTime is required for jobs with terminal %s=True condition", cType)))
+				}
+				if job.Status.Active > 0 {
+					allErrs = append(allErrs, field.Invalid(statusFld.Child("active"), job.Status.Active, fmt.Sprintf("active>0 is invalid for jobs with terminal %s=True condition", cType)))
+				}
+				if ptr.Deref(job.Status.Terminating, 0) > 0 {
+					allErrs = append(allErrs, field.Invalid(statusFld.Child("terminating"), *job.Status.Terminating, fmt.Sprintf("terminating>0 is invalid for jobs in terminal %s=True condition", cType)))
+				}
+				if job.Status.UncountedTerminatedPods != nil && len(job.Status.UncountedTerminatedPods.Failed)+len(job.Status.UncountedTerminatedPods.Succeeded) > 0 {
+					allErrs = append(allErrs, field.Invalid(statusFld.Child("uncountedTerminatedPods"), job.Status.UncountedTerminatedPods, fmt.Sprintf("uncountedTerminatedPods needs to be empty for jobs in terminal %s=True condition", cType)))
+				}
+			}
+		}
+		// validation to make sure Complete=True does not occur with Failure conditions
+		for _, failureType := range []batch.JobConditionType{batch.JobFailed, batch.JobFailureTarget} {
```

Comment:
If I have JobFailureTarget set to true, am I allowed to set Completed true?

Reply:
nope